tests/src/gui/testqgsdatetimeedit.cpp

```diff
@@ -226,5 +235,31 @@ void TestQgsDateTimeEdit::focus()
   QCOMPARE( dateedit3->text(), QStringLiteral( "nope" ) );
 }
 
+void TestQgsDateTimeEdit::testDateTime()
+{
+  QgsApplication::setNullRepresentation( QString( "nope" ) );
+  QWidget *w = new QWidget(); //required for focus events
```

Comment:
Isn't this leaked?
I think it's good practice to not leak in tests, so they can be run through valgrind to find issues.

Reply:
Good catch, I just copied that init code from the existing test case a few lines above and focused on the test logic, funny that it escaped from two reviewers: https://github.com/qgis/QGIS/pull/32158, you really do have an eye for this!